src/dialog.jsx

```diff
@@ -379,27 +375,30 @@ let Dialog = React.createClass({
     }
   },
 
-  _onShow() {
-    if (this.props.onShow) this.props.onShow();
+  _show() {
+    this.refs.dialogOverlay.preventScrolling();
+    this.setState({ open: true });
+  },
+
+  _dismiss() {
+    CssEvent.onTransitionEnd(ReactDOM.findDOMNode(this), () => {
+      this.refs.dialogOverlay.allowScrolling();
+    });
+
+    this.setState({ open: false });
   },
 
-  _onDismiss() {
-    if (this.props.onDismiss) this.props.onDismiss();
+  _requestClose() {
+    if (this.props.onRequestClose) this.props.onRequestClose();
```

Comment:
With an uncontrolled dialog (`isOpen===null`) I think that we should hide the dialog without triggering `onRequestClose`